typedoc.json

```diff
@@ -0,0 +1,7 @@
+{
+    "entryPoints": ["index.ts"],
+    "out": "docs/api-ref",
+    "name": "Voiceflow Runtime Client JS",
```

Comment:
`typedoc` turned out to be a little too inflexible and probably wouldn't have provided as much as value compared to if manually wrote out some of the API reference. 

For example, `typedoc` doesn't express complex user-defined types as primitive JavaScript types, you get something like:
```ts
const SomeTrace = {
    x: SomeSubTypeA;
    y: SomeSubTypeB;
    z: MoreSubTypes;
};
```
This isn't really helpful because the user will need to look up `SomeSubTypeA`, which might also be defined in terms of user-defined types. So to make `SomeSubTypeA` show up typedoc, we need to export more types, maybe some from `general-types`. Thus, we just end up having to export 100s of different types just to document something simple.

There's a probably a plugin or some trick we can use to solve this, but I don't think there's enough time this sprint to figure out typedoc and make it usable for other people.

I'm just going to leave this here in case we want to revisit `typedoc` in the future.